src/Avalonia.Controls/Notifications/WindowNotificationManager.cs

```diff
@@ -0,0 +1,170 @@
+using System;
+using System.Collections;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using Avalonia.Controls.Primitives;
+using Avalonia.Interactivity;
+using Avalonia.VisualTree;
+
+namespace Avalonia.Controls.Notifications
+{
+    public class WindowNotificationManager : TemplatedControl, IManagedNotificationManager
+    {
+        private IList _items;
+
+        public static NotificationPosition GetPosition(TopLevel obj)
+        {
+            return obj.GetValue(PositionProperty);
+        }
+
+        public static void SetPosition(TopLevel obj, NotificationPosition value)
+        {
+            obj.SetValue(PositionProperty, value);
+        }
+
+        public static readonly AvaloniaProperty<NotificationPosition> PositionProperty =
+          AvaloniaProperty.RegisterAttached<WindowNotificationManager, TopLevel, NotificationPosition>("Position", defaultValue: NotificationPosition.TopLeft, inherits: true);
+
+        public NotificationPosition Position
+        {
+            get { return GetValue(PositionProperty); }
+            set { SetValue(PositionProperty, value); }
+        }
+
+        public static int GetMaxItems(TopLevel obj)
+        {
+            return obj.GetValue(MaxItemsProperty);
+        }
+
+        public static void SetMaxItems(TopLevel obj, int value)
+        {
+            obj.SetValue(MaxItemsProperty, value);
+        }
+
+        public static readonly AvaloniaProperty<int> MaxItemsProperty =
+          AvaloniaProperty.RegisterAttached<WindowNotificationManager, TopLevel, int>("Position", defaultValue: 5, inherits: true);
+
+        public int MaxItems
+        {
+            get { return GetValue(MaxItemsProperty); }
+            set { SetValue(MaxItemsProperty, value); }
+        }
+
+        public WindowNotificationManager(Window host)
+        {
+            if (VisualChildren.Count != 0)
+            {
+                Install(host);
+            }
+            else
+            {
+                IDisposable stopListening = null;
+
+                stopListening = Observable.FromEventPattern<TemplateAppliedEventArgs>(host, nameof(host.TemplateApplied))
```

Comment:
Rather than disposing of the subscription in the subscription, you can just use `Take(1)` to make sure that the subscription is automatically disposed after the first time it fires.

Reply:
done